keps/sig-network/2943-networkpolicy-status/README.md

```diff
@@ -0,0 +1,464 @@
+name as `myprovider.com/PartialFailure?`
+
+Also, what kind of polatiry should we use in a default condition Type? Positive polarity 
+like "Accepted=bool" or negative polarity like PartialFailure=true?
+
+See the original discussion in https://github.com/kubernetes/enhancements/pull/2947#discussion_r796974848
+<<[/UNRESOLVED]>>
+```
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+Unit tests:
+* test API validation logic
+* test API strategy to ensure disabled fields
+* test random insertions into the conditions field
+
+E2E tests:
+* Add e2e tests exercising only the API operations for the policy condition fields
+
+### Graduation Criteria
+
+#### Alpha
+
+- Add a feature gated new status subresource to NetworkPolicy
+- Initial e2e tests completed
+- Communicate Network Policy providers about the new field
+- Add validation tests in API
+
+#### Beta
+
+- Status has been supported for at least 1 minor release
+- **two** commonly used NetworkPolicy providers implement the new field,
+with generally positive feedback on its usage.
+- Feature Gate is enabled by Default.
+
+#### GA
+
+- At least **three** NetworkPolicy providers support the status subresource and the conditions
+- Status has been enabled by default for at least 2 minor release
+
+### Upgrade / Downgrade Strategy
+
+If upgraded no impact should happen as this is a new field.
+
+If downgraded the Network Policy provider wont be able to write into 
+this new field, which may lead to errors and inconsistency.
+
+The provider implementing this feature should deal with errors 
+returned from the API Server and bypass them in case of the field not existing in the current version.
+
+### Version Skew Strategy
+
+N/A
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+-->
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: NetworkPolicyStatus
+  - Components depending on the feature gate:
+    - API Server
+    - Network Policy Providers
+
+
+###### Does enabling the feature change any default behavior?
+
+No
```

Comment:
I just realized that it actually should be "yes", because after enabling the feature gate network policy providers will start setting some status.

Reply:
They will just if implemented on their side. This is why I thought this was no. The change of a "default behavior" for me would be something like "enabling the feature gate will make all TCP connections deny something".

But let me change here :)